lib/cocoapods/target/pod_target.rb

```diff
@@ -184,6 +184,15 @@ def dependencies
       end.uniq
     end
 
+    # @return [Array<PodTarget>] the recursive targets that this target has a
+    #         dependency upon.
+    #
+    def recursive_dependent_targets
+      targets = dependent_targets + dependent_targets.flat_map(&:dependent_targets)
```

Comment:
Doesn't this stop the recursion at one level only? I'd have though the right recursive implementation should be `dependent_targets + dependent_targets.flat_map(&:recursive_dependent_targets)`?